aten/src/ATen/native/cuda/Distributions.cu

```diff
@@ -724,13 +724,34 @@ Tensor& normal_out_cuda(Tensor& output, double mean, const Tensor& std, Generato
 }
 
 Tensor& normal_out_cuda(Tensor& output, const Tensor& mean, const Tensor& std, Generator* gen) {
+  bool expandable = are_expandable(mean.sizes(), std.sizes());
+  // allow resize if output is an empty tensor
+  if (output.sizes().equals({0})) {
+    if (expandable) {
+      auto shape = at::infer_size(mean.sizes(), std.sizes());
+      at::native::resize_(output, shape, LEGACY_CONTIGUOUS_MEMORY_FORMAT);
+    }
+    else {
+      at::native::resize_(output, mean.sizes(), LEGACY_CONTIGUOUS_MEMORY_FORMAT);
```

Comment:
You don't want to resize the output if the op isn't going to succeed because resize is a destructive operation.  It is true that the previous would potentially write data and then fail, but it wouldn't resize.

Reply:
Will think about this more. Probably create a temp tensor doing the op in this case. And if success, then doing a inplace copy to the output tensor.

Reply:
I don't think this is correct.  We define out variants of this function, so you want to get the resize behavior correct.

I don't follow the argument about the non-out variant either.  What's the point of passing it to normal_cuda anyway -- you know it will fail later.